packages/sanes-chrome-extension/src/routes/tx-request/components/ShowRequest/index.tsx

```diff
@@ -14,33 +15,43 @@ interface Props {
   readonly showRejectView: () => void;
   readonly sender: string;
   readonly creator: string;
-  readonly tx: SendTransaction;
+  readonly tx: SupportedTransaction;
 }
 
-const Layout = ({ sender, tx, creator, onAcceptRequest, showRejectView }: Props): JSX.Element => (
-  <PageLayout id={TX_REQUEST_SHOW} color="white" primaryTitle="Tx" title="Request">
-    <Block textAlign="center" marginBottom={2}>
-      <Typography variant="body1" inline>
-        {'The following site: '}
-      </Typography>
-      <Typography variant="body1" color="primary" inline>
-        {sender}
-      </Typography>
-      <Typography variant="body1" inline>
-        {' wants you to sign:'}
-      </Typography>
-    </Block>
-    <Req tx={tx} creator={creator} />
-    <Block marginBottom={3} />
-    <Button variant="contained" fullWidth onClick={onAcceptRequest}>
-      Approve
-    </Button>
-    <Block marginTop={2} />
-    <Button variant="contained" fullWidth color="secondary" onClick={showRejectView}>
-      Reject
-    </Button>
-    <Block marginBottom={2} />
-  </PageLayout>
-);
+const Layout = ({ sender, tx, creator, onAcceptRequest, showRejectView }: Props): JSX.Element => {
+  let req: JSX.Element;
+
+  if (isSendTransaction(tx)) {
+    req = <ReqSendTransaction tx={tx} creator={creator} />;
+  } else {
+    throw new Error('Received transaction type that cannot be displayed');
```

Comment:
This will make extension to crash if we will not implement "[Error Boundaries](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html#introducing-error-boundaries)" for the extension.

Reply:
Crashing is perfect in this case. Incoming transactions with unsupported type are already rejected by the signing server: https://github.com/iov-one/ponferrada/blob/5fd61f2/packages/sanes-chrome-extension/src/extension/background/model/signingServer/index.ts#L67-L69 If we receive an unsupported transaction type in this part of the code, this is a programming error.

Reply:
So we need to implement "Error Boundaries" to receive reports to Sentry about this. @kmw101 what do you think?

Reply:
"Error Boundaries" are an interesting concept. Thanks for the link. However, I tend to disagree with

> A JavaScript error in a part of the UI shouldn’t break the whole app.

If there is an unhandled exception, this means no developer thought about this error case before. Ignoring such an exception and continuing to run the app can lead to invalid app state and undetected errors. Letting unhandled exceptions crash the application increase chances that the programming error is detected during development or when running tests.

I think we should just crash the application for now and add Sentry later on as an additional way to detect and understand errors that happen on the user's machine and did not occur during development.

Reply:
Anyway we need to create issue for this case. Let's wait what @kmw101 will tell about this.

Reply:
I tend to agree with the point about letting unhandled exceptions crash, I prefer that to something silently failing and nobody knows or can find out what happened.

Let's keep this simple and if we have time later publish to Sentry.

Reply:
Added an issue: https://github.com/iov-one/ponferrada/issues/347